pygments/lexers/gsql.py

```diff
@@ -0,0 +1,78 @@
+import re
+
+from pygments.lexer import RegexLexer, include, bygroups, using, this
+from pygments.token import Keyword, Punctuation, Comment, Operator, Name,\
+    String, Number, Whitespace
+
+
+__all__ = ["GSQLLexer"]
+
+class GSQLLexer(RegexLexer):
+    name = 'GSQL'
+    aliases = ['gsql']
+    filenames = ['*.gsql']
+
+    flags = re.MULTILINE | re.IGNORECASE
+
+    tokens = {
+        'root': [
+            include('comment'),
+            include('keywords'),
+            include('clauses'),
+            include('accums'),
+            include('relations'),
+            include('strings'),
+            include('whitespace'),
+            include('barewords'),
+            include('operators'),
+        ],
+        'comment': [
+            (r'.*\#.*\n', Comment.Single),
+            (r'.*\/\*\s*.*\s*\*\/', Comment.Multiline),
+        ],
+        'keywords': [
+            (r'(ACCUM|AND|ANY|API|AS|ASC|AVG|BAG|BATCH|BETWEEN|BOOL|BOTH|'
```

Comment:
Please use the `words` function for long lists of keywords which can optimize the matching, see for instance: https://github.com/pygments/pygments/blob/master/pygments/lexers/python.py#L202